sound/soc/intel/boards/ehl_rt5660.c

```diff
@@ -0,0 +1,387 @@
+// SPDX-License-Identifier: GPL-2.0
+
+/*
+ *  Intel Elkhart Lake I2S Machine driver with RT5660 Codec
+ *
+ *  Copyright (C) 2019 Intel Corp
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; version 2 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful, but
+ *  WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  General Public License for more details.
+ *
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ */
+#include <linux/module.h>
+#include <linux/acpi.h>
+#include <sound/core.h>
+#include <sound/jack.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/soc-acpi.h>
+
+#include "../../codecs/rt5660.h"
+#include "../../codecs/hdac_hdmi.h"
+
+#define EHL_BE_FIXUP_RATE	48000
+#define RT5660_CODEC_DAI	"rt5660-aif1"
+#define DUAL_CHANNEL 2
+
+struct ehl_card_private {
+	struct list_head hdmi_pcm_list;
+};
+
+static const struct snd_kcontrol_new ehl_controls[] = {
```

Comment:
remove all elh prefix, it's not good for code reuse.

Reply:
OK

Reply:
@plbossart I've gone through a couple iterations with this. At some points of time I was happy with identical names in similar files, e.g. codec or camera drivers, but then I realised, that that makes searching more difficult. And makes the code more error prone when bad copy-paste chunks become more difficult to identify. So, currently I actually prefer different names in different files. But I do see, that SOF and possibly ALSA prefer identical names... I can certainly live with that too, just makes grepping / tagging more difficult, but maybe we could consider breaking this convention...

Reply:
yes, you've mentioned this before. It's a valid point, but that lead us to using bxt_da7219 prefixed variables and functions for GLK and CML. So if we plan to reuse code across platforms, it's nonsensical to have a platform-specific prefix.